meeshkan/serve/mock/faker/data_faker.py

```diff
@@ -0,0 +1,50 @@
+from faker import Faker
+from http_types import Request
+from openapi_typed_2 import OpenAPIObject, Any, Operation
+
+from meeshkan.serve.mock.faker.schema_faker import MeeshkanSchemaFaker
+from meeshkan.serve.mock.storage import Storage
+
+
+
+class MeeshkanDataFaker(MeeshkanSchemaFaker):
+    def __init__(self, text_faker: Faker, request: Request, spec: OpenAPIObject, storage: Storage):
+        super().__init__(text_faker, request, spec, storage)
+
+    def fake_ref(self, schema: Any, depth: int):
+        name = schema["$ref"].split("/")[2]
+        if name in self._storage:
+            if name == self._entity_name and self._generated_data is not None:
+                return self._generated_data
+            else:
+                return self._storage.query_one(name, self._request)
+        else:
+            return self.fake_it(self._top_schema["definitions"][name], depth)
+
+    def fake_ref_array(self, schema: Any, depth: int, count: int):
+        name = schema["$ref"].split("/")[2]
+        if name in self._storage:
+            return self._storage.query(name, self._request)
+        else:
+            return [self.fake_it(self._top_schema["definitions"][name], depth) for _ in range(count)]
+
+    def _update_data(self, method: Operation):
+        if method._x is None or 'x-meeshkan-operation' not in method._x:
```

Comment:
I think this shows one difficulty in putting `x-meeshkan` stuff inside OpenAPI as the logic of how such extensions are handled is deep inside the faker and not a stand-alone "extension logic module". I'm a bit worried this will prove hard to maintain and debug. I don't have a better solution as I don't understand that well everything is thing is trying to achieve but just a comment.

Reply:
That fields are optional even for the new approach. They are filled only when they can be filled. So it should support specs without that fields in any case. But the checks are massive. probably, we could inherit and extend spec somehow to reduce it to `if method.meeshkan_opertion is not None`.